### ISSUE TITLE
devnet bricks when instantiating contract on aarch64

### ISSUE THREAD

**Issue by cor** [unionlabs/union#3270]
### 1. start devnet-uniond

```sh
nix run .#devnet-uniond -L
```

### 2. build union home

```sh
nix build .#union-home
```

### 3. upload contract

I've put `ethereum_light_client.wasm` in a sibling folder `test-contracts`, but you can place it wherever. You can [download ethereum_light_client.wasm here](https://pub-7a09c5b75b8b40d89aedfa4eb65ab1eb.r2.dev/ethereum_light_client.wasm).

```sh
nix run .#uniond -- tx wasm store ../test-contracts/ethereum_light_client.wasm --from alice --home ./result --keyring-backend test --gas auto --gas-adjustment 1.4 --node http://127.0.0.1:26657 --chain-id union-devnet-1
```

### 4. instantiate contract

```sh
nix run .#uniond -- tx wasm instantiate 1 '{}' --node http://127.0.0.1:26657 --home result --from alice --keyring-backend test --label union-ibc-2 --admin union1jk9psyhvgkrt2cumz8eytll2244m2nnz4yt2g2 --chain-id union-devnet-1
```

### 5. devnet is bricked

You should see that RAM usage keeps growing, and that the devnet no longer produces new blocks.

**Comment by cor** [comment]
![D7B11FA1-FCE5-4E0C-88C7-024F908ED246](https://github.com/user-attachments/assets/77bddd9f-f44e-4b8d-8589-4d06479e713f)

After instantiating, all `uniond` processes take 100% of one CPU core.

![image](https://github.com/user-attachments/assets/56628b75-2110-4454-8bfa-9e26fe490be4)

when killing the `devnet-union` with `C-c`, RAM usage drops from 120GB to 1GB

**Comment by cor** [comment]
We profiled what happens after the `wasm instantiate` call, and it is clear that the actual processing of the tx itself is causing the crash:

![profile008](https://github.com/user-attachments/assets/620dfacc-55a9-4336-9eaf-73ce2b0fdefe)

(right click and open in new tab ^)

**Comment by cor** [comment]
<img width="555" alt="Screenshot 2024-11-18 at 13 20 36" src="https://github.com/user-attachments/assets/1354fd95-28de-4acf-a6e9-86ce0c21483e">
(@aeryz)

**Comment by aeryz** [comment]
Example issue:
https://github.com/confluentinc/confluent-kafka-go/issues/248

**Comment by cor** [comment]
<img width="1010" alt="image" src="https://github.com/user-attachments/assets/0915b279-93cb-497a-a8bd-893f04763987">
interestingly, when we ran the same experiment again, it hung on a different final call

**Comment by hussein-aitlahcen** [comment]
Note: CPU usage is getting high on nodes where the TX are being broadcasted

**Comment by PoisonPhang** [comment]
Closed by #3277 ?
